File: REST API/C-Sharp/ReplyTicket.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Threading.Tasks;

public class ReplyTicket
{
    static HttpClient client = new HttpClient(); // One HttpClient instance be reused.
                                                 //HttpClient is used to make WebAPI Request from console application.

    static async Task Main(string[] args)
    {
        string domain = "YOUR_DOMAIN"; //Your bolddesk domain name.

        string apiPath = "/api/v1/tickets/{ticketId}/updates"; //Your API path.

        string apiKey = "YOUR_API_KEY"; //Your API key.

        client.BaseAddress = new Uri("https://" + domain); //Generate API path to call.

        client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json")); //Header Value for Content type.

        client.DefaultRequestHeaders.Add("x-api-key", apiKey); //For APIKEY Authentication.

        try
        {
            //Post data to reply ticket.
            var ticketData = new TicketUpdate    //Sample Data's to reply ticket.
            {
                Description = "sample description",
                Attachments = "tokenValue_1, tokenValue_2",
                AtMentionUserIds = new List<int> { 1, 2 },
                lastUpdateID = 12,
                lastRefreshDate = "2022-02-21T08:04:19.785Z",
                TimeSpent = 5,
                MessageTags = "tag1,tag2",
                UpdatedByUserIdOrEmailId = "12",
                SkimEmailNotification = true,
                TicketStatusId = 1,
                Cc = new List<int> { 1, 2 },
                ReplyOnBehalfOfRequester = false,
                DontAppendOnBehalfOfRequesterMessage = true
            };

            HttpResponseMessage response = await client.PostAsJsonAsync(apiPath, ticketData).ConfigureAwait(false);//To send a POST request as an asynchronous operation to the specified Uri with the given value serialized as JSON.

            var responseData = await response.Content.ReadAsStringAsync().ConfigureAwait(false); //To read the response.

            response.EnsureSuccessStatusCode(); //For Confirms whether the response is success or Not.

            Console.WriteLine(responseData); //Return the response.
        }

        //To catch the error message.
        catch (Exception ex)
        {
            Console.WriteLine("ERROR: ");
            Console.WriteLine("Status Code: " + ex.StatusCode); //Returns the error status.
            Console.WriteLine(ex.Message);//Returns the error message.
            Console.WriteLine(ex.StackTrace);//Returns the stack at the time uncaught exception occurs.            
        }

    }
}

public class TicketUpdate
{
    public string Description { get; set; } = string.Empty;

    public string Attachments { get; set; }

    public List<int> AtMentionUserIds { get; set; }

    public long lastUpdateID { get; set; }

    public string lastRefreshDate { get; set; }

    public int TimeSpent { get; set; }

    public string MessageTags { get; set; }

    public string UpdatedByUserIdOrEmailId { get; set; }

    public bool SkimEmailNotification { get; set; }

    public int TicketStatusId { get; set; }

    public List<int> Cc { get; set; }

    public bool ReplyOnBehalfOfRequester { get; set; }

    public bool DontAppendOnBehalfOfRequesterMessage { get; set; }
}

File: REST-API-Python/reply_ticket.py
import requests
import traceback

# One Session instance should be reused (similar to a single HttpClient instance).
session = requests.Session()


def main():
    domain = "YOUR_DOMAIN"         # Your bolddesk domain name (e.g., "yourcompany.bolddesk.com")
    api_key = "YOUR_API_KEY"       # Your API key
    ticket_id = "YOUR_TICKET_ID"   # <-- Replace with actual ticketId (e.g., "12345")

    # Your API path (same as C#): /api/v1/tickets/{ticketId}/updates
    api_path = f"/api/v1/tickets/{ticket_id}/updates"

    base_url = f"https://{domain}".rstrip("/")
    url = f"{base_url}{api_path}"

    # Header Value for Content type + APIKEY Authentication (matches your C# headers)
    session.headers.update({
        "Accept": "application/json",
        "x-api-key": api_key
    })

    try:
        # Post data to reply ticket (sample data)
        #
        # NOTE on JSON casing:
        # C# System.Text.Json commonly serializes PascalCase properties as camelCase by default.
        # So Description -> description, ReplyOnBehalfOfRequester -> replyOnBehalfOfRequester
        ticket_data = {
            "description": "sample description",
            "replyOnBehalfOfRequester": False
        }

        # Send POST request with JSON body (equivalent to PostAsJsonAsync)
        response = session.post(url, json=ticket_data)

        # Read response text (like ReadAsStringAsync)
        response_text = response.text

        # Ensure success (like EnsureSuccessStatusCode)
        response.raise_for_status()

        # Print response
        print(response_text)

        # If you prefer JSON output (only if API returns JSON), use:
        # print(response.json())

    except requests.exceptions.HTTPError as ex:
        status_code = ex.response.status_code if ex.response is not None else None
        body = ex.response.text if ex.response is not None else ""
        print("ERROR:")
        print("Status Code:", status_code)
        print("Message:", str(ex))
        print("Response Body:", body)
        print("StackTrace:")
        print(traceback.format_exc())

    except Exception as ex:
        print("ERROR:")
        print("Status Code: N/A")
        print("Message:", str(ex))
        print("StackTrace:")
        print(traceback.format_exc())


if __name__ == "__main__":
    main()